app/knigts.py

```diff
@@ -0,0 +1,43 @@
+from __future__ import annotations
+
+
+class Knight:
+    def __init__(self, name: str, power: int, hp: int) -> None:
+        self.name = name
+        self.power = power
+        self.hp = hp
+        self.protection = 0
+
+    def apply_armour(self, armour: list[dict]) -> None:

```

Comment:
Use 'if armour:' instead of 'if armour is not []:' to check if the list is not empty 📝